project/Boilerplate.scala

```diff
@@ -270,13 +272,18 @@ object Boilerplate {
         }
       val outputType = if (arity != 1) s"Product$arity[${`A..N`}]" else `A..N`
 
+      val deprecate = if (arity != 1)
+        s"""@deprecated(message="does not work correctly for Scala 3. Prefer forTupleProduct$arity", since="0.14.6")"""
+      else ""
+
```

Comment:
I don't want to deprecate this.  It works just fine for many existing workflows, just not the `Tuple.fromProductTyped`